### PR TITLE
[Azure.Core] Update Tests for ModelSerializer

### PR DESCRIPTION
**Pull request by nisha-bhatia**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

**Comment by nisha-bhatia** [comment]
This PR addresses the following tests from https://github.com/Azure/azure-sdk-for-net/issues/37290

Scenarios for testing serializer – do for both Json and XML

- [ ]  Models with reference to itself
- [x]  • Model with null property
- [ ]  • BYOM - Dynamic model testing? Model with a type T model
- [ ]  • Can we have a model with an XML model property? Eg. A Json model with a Json model property
- [x]  • Models with list of models
- [x]  • List with primitive properties
- [x]  • Model with readonly property
- [ ]  • Discriminated set model
- [ ]  o Deserialize and serialize using the base model and have the expected value be equal to the serialization of the concrete value
- [ ]  o Deserialize as concrete value and return the base model
- [x]  • Model with primitive properties
- [ ]  • Model with a json converter attribute
- [ ]  o Target serialization is the json converted model
- [ ]  o Json converted model is a property of the target serialization
- [x]  • Model with a dictionary
- [x]  • Testcase with an additional property
- [ ]  o Test at different layers
- [ ]  o Check in target of the serialization
- [ ]  o Property model of the target of the serialization
- [ ]  o Lands inside Json converted model

**Comment by nisha-bhatia** [comment]
Replaces: https://github.com/Azure/azure-sdk-for-net/pull/38117

File: sdk/core/Azure.Core/tests/public/ModelSerializationTests/ModelXTests.cs [hunk 1]
```diff
@@ -37,6 +44,8 @@ protected override void CompareModels(ModelX model, ModelX model2, ModelSerializ
         protected override string GetExpectedResult(ModelSerializerFormat format)
         {
             string expected = "{\"kind\":\"X\",\"name\":\"xmodel\"";
+            expected += ",\"fields\":[\"testField\"]";
```
Comment (m-nash) [inline sdk/core/Azure.Core/tests/public/ModelSerializationTests/ModelXTests.cs:47]: we talked about having a convention for all of these strings.  Something along the lines of

Model/ModelWire.json
Model/ModelJson.json
Model/ExpectedWire.json
Model/ExpectedJson.json

From this convention the ModelTests base class could simply load all of those files and do all comparisons.

Did we decide to punt this?  If so can you open a tracking issue so we don't lose track?

Comment (nisha-bhatia) [reply sdk/core/Azure.Core/tests/public/ModelSerializationTests/ModelXTests.cs:47]: I wasn't aware that we had decided on this convention - I made an issue to track the work: https://github.com/Azure/azure-sdk-for-net/issues/38404

**Review by m-nash — APPROVED** [review]

**Pull request merged**